<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <xacro:macro name="cia402_ros2_control_test" params="
      name
      prefix
      bus_config
      master_config
      can_interface_name
      master_bin">

        <ros2_control name="${name}" type="system">
            <hardware>
              <plugin>canopen_ros2_control/Cia402System</plugin>
              <param name="bus_config">${bus_config}</param>
              <param name="master_config">${master_config}</param>
              <param name="can_interface_name">${can_interface_name}</param>
              <param name="master_bin">"${master_bin}"</param>
            </hardware>
            <joint name="${prefix}node_1">
                <param name="node_id">2</param>
            </joint>
<!--            <joint name="${prefix}node_2">-->
<!--                <param name="node_id">3</param>-->
<!--            </joint>-->
        </ros2_control>

    </xacro:macro>

</robot>
